proto/streamlit/proto/AutoRerun.proto

```diff
@@ -0,0 +1,25 @@
+/**!
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ syntax = "proto3";
+
+ message AutoRerun {
+   // The interval of reruns in seconds
+   float interval = 1;
+
+   // Optional: the fragment ID to rerun
```

Comment:
It would take a bit of work to have this enable a native version of the autorefresh library (we'd mostly need to figure out how to expose this to developers + remove an assumption or two in the frontend that assumes autorerun is always associated with a fragment), but we may want to do this as a small project in the future. For now, I'll remove the optional since right now it's a bit misleading